### PR TITLE
Hint how to set axis limits with margins

### PR DESCRIPTION
**Pull request by dlukes**
## PR summary

I recently came across a use case where I wanted to manually tweak axis limits, but also apply the margins which are added when autoscaling. It took me a while to figure out how to do this in an elegant way that would avoid computing the margins manually (in order to defer to Matplotlib for this): by modifying the `Axes.dataLim` bounding box and re-applying autoscaling. Since `Axes.set_{x,y}lim` is the most straightforward starting point when considering axis limits tweaks, I think it would make sense to document this approach there, as a helpful hint to anyone trying to do something similar.

I understand this might not be wanted for a variety of reasons, or perhaps you'd rather put this somewhere else than in the API docs for the `Axes.set_{x,y}lim` functions, or perhaps you'd just phrase it differently. I just wanted to start the conversation over a concrete suggestion; that's why a PR instead of just opening an issue. I do think that it would be useful to document this somewhere, preferably in a way that's reachable from the `Axes.set_{x,y}lim` docs, but I'm not too particular about the details and would be happy to take suggestions about places which might be a better fit.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

**Comment by timhoffm** [comment]
Thanks for the contribution.

A textual description in the docstring is quite hard to follow. I think there are two alternatives:

Likely the right one: Create an example.

Maybe (but likely not - that would need more thorough thought and discussion): Expand the `set_xlim()` API with margin capability.

**Comment by rcomer** [comment]
This one has stalled for some time, so let's close it.  An example can always be added in a new PR.
